### ISSUE TITLE
Try using a network pretrained using Resnet 50

### ISSUE THREAD

**Issue by vikasmahato** [marco-c/autowebcompat#112]
References #7 

This paper provides various benchmarks for pretrained models used for transfer learning.
https://openreview.net/pdf?id=Bygq-H9eg

It also suggests that using Resnet should perform better as compared to VGG.

**Comment by vrishank97** [comment]
Hi. I'm new here. I'd like to work on this issue.

**Comment by marco-c** [comment]
There are already two PRs about this: https://github.com/marco-c/autowebcompat/pull/61 and https://github.com/marco-c/autowebcompat/pull/105.

**Comment by marco-c** [comment]
N.B.: They are adding the architecture, but nothing regarding a pretrained network

**Comment by vrishank97** [comment]
Great. Should I add the new model with imagenet weights in network.py ?

**Comment by vrishank97** [comment]
Great. Should I work on this in network.py ?

**Comment by marco-c** [comment]
Yes, the first step would be to figure out how to use an already pretrained network, since the size of our images are different than the default VGG, Resnet, etc.

**Comment by vrishank97** [comment]
I'm considering squashing images to 224x224. As we are mainly concerned with UI/UX elements and not text, I don't think squashing will affect performance.

**Comment by marco-c** [comment]
Yeah, that's one of the option that we should try

**Comment by vikasmahato** [comment]
@vrishank97 I was working on this issue. However if you want to take it please let me know.

**Comment by vrishank97** [comment]
@vikasmahato Since we are just experimenting with different ways of getting the transfer learning to work, we'll get results faster if we work in parallel. I will try downscaling images to imagenet dimensions and re-training resnet 50 on it. What approach are you currently working on?

**Comment by vikasmahato** [comment]
@vrishank97 I was also thinking the same. However since you are already doing it I'll try to use Inception V3 and see how it performs.

**Comment by vrishank97** [comment]
@vikasmahato Great. Can you try using the Inception-Resnet-v2 instead? It had a higher accuracy on imagenet. 

https://research.googleblog.com/2016/08/improving-inception-and-image.html

**Comment by vikasmahato** [comment]
@vrishank97 Sure!

**Comment by Shashi456** [comment]
@vrishank97 @vikasmahato I wanted to ask you both what are the steps in using any pretrained network , are you finetuning the last layer ?

**Comment by vrishank97** [comment]
We freeze the convo layers and retrain the fully connected layers with a custom softmax output layer.

**Comment by vrishank97** [comment]
Here are some resources
https://blog.keras.io/building-powerful-image-classification-models-using-very-little-data.html
https://towardsdatascience.com/transfer-learning-using-keras-d804b2e04ef8

**Comment by marco-c** [comment]
We should try both approaches: freeze all except the top one and also keep training everything.

**Comment by vrishank97** [comment]
@marco-c The dimensions shouldn't be an issue for transfer learning. All images are resized by prepare_images() in utils.py

**Comment by marco-c** [comment]
> @marco-c The dimensions shouldn't be an issue for transfer learning. All images are resized by prepare_images() in utils.py. We only need to load imagenet weights.

Yes, but by resizing we might be losing some information.

**Comment by vrishank97** [comment]
Agreed. But I think its mainly the text areas where we lose information, major UI elements would still be recognisable, especially if we use Inception. It has an input of 299x299 instead of 224x224, so lower information loss.

**Comment by marco-c** [comment]
Yes, hopefully yes.

**Comment by marco-c** [comment]
Closing in favor of https://github.com/marco-c/autowebcompat/issues/194.
